app/views/devise/registrations/edit_password.html.erb

```diff
@@ -16,6 +16,7 @@
     } %>
 
     <%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do %>
+    <%= email_field_tag 'email', resource.email, class: "govuk-!-display-none", autocomplete: "off", "aria-hidden": true %>
```

Comment:
shouldn't this be type hidden rather than hidden by css?

Reply:
The reason being - the browser validation will trigger and the user will not be able to submit the form.

Reply:
That's a good question. We've been going off the guidance [here](https://www.chromium.org/developers/design-documents/create-amazing-password-forms), which states that the hidden input should have `type="text"` and be hidden with CSS. 
It seems that using `type="hidden"` means it won't be recognised by some browsers (namely Chrome, [from what I could see](https://stackoverflow.com/questions/48525114/chrome-warning-dom-password-forms-should-have-optionally-hidden-username-fi/62139003)). 
Annoyingly, Chrome also seems to ignore `autocomplete="off"` entirely.

Reply:
> The reason being - the browser validation will trigger and the user will not be able to submit the form if for some reason the email is not valid.

Ah so weird, this comment didn't show up for me earlier. 

In any case, this is the email they would have already entered on the previous page. So it would have already been validated in the previous step... unless I'm misunderstanding what you mean.

Reply:
> Annoyingly, Chrome also seems to ignore autocomplete="off" entirely.

Does this mean there's a possibility that a password manager might update this hidden field?

Reply:
@domoscargin I would have thought so too, but it seems like browser autofill and LastPass (at least for me?) are ignoring that field. I can't say I totally understand how/why it works, but it seems to work. 

It's just an assumption, but perhaps password managers/autofill don't update hidden inputs as a means to prevent phishing via autofill?

Reply:
I think if the hidden fields are being used purely as a means to get autofill playing nicely and not actually being used for backend validation, this is fine.

`type="hidden"` may be an issue because the attribute is wearing the "autofill anchor mantle": https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#attr-fe-autocomplete so `autocomplete="off"` is not allowed, and I guess it's being interpreted by browsers/password managers in a different way.